backend/LeetWars.Core/LeetWars.Core.BLL/Services/ChallengeService.cs

```diff
@@ -245,6 +245,44 @@ public async Task DeleteChallengeAsync(long challengeId)
             await _context.SaveChangesAsync();
         }
 
+        public async Task SetWeeklyChallenges()
+        {
+            await ResetLastWeeklyChallenge();
+
+            var levels = Enum.GetValues(typeof(LanguageLevel))
+                                            .Cast<LanguageLevel>()
+                                            .ToArray();
+
+            foreach(var level in levels) 
+            {
+                var challengesByLevel = _context.Challenges
+                    .Where(x => x.Level != null && x.Level.SkillLevel == level)
+                    .AsQueryable();
+
+                var randomPosition = GetRandomInt(challengesByLevel.Count());
```

Comment:
```suggestion
                var randomPosition = GetRandomInt(challengesByLevel.CountAsync());
```

Reply:
GetRandomInt is noe anync method
![image](https://github.com/BinaryStudioAcademy/bsa-2023-leetwars/assets/68563967/c23effa9-1971-4104-8b3e-c082351a429e)

```
 private static int GetRandomInt(int maxValue)
 {
     if (maxValue == 0)
     {
         return 0;
     }

     using (var generator = RandomNumberGenerator.Create())
     {
         var data = new byte[4];
         generator.GetBytes(data);
         var randomValue = Math.Abs(BitConverter.ToInt32(data, 0));

         return randomValue % maxValue;
     }
 }
```

Reply:
We don't care about GetRandomInt. You need to await CountAsync result

Reply:
aaa got it